ContentPatcher/ModEntry.cs

```diff
@@ -463,5 +492,71 @@ select value.Trim()
+        {
+            if (configField.AllowValues.Any())
+            {
+                if (configField.AllowMultiple)
+                {
+                    // Whitelist + multiple options = fake with multiple checkboxes
+                    foreach (string allowedValue in configField.AllowValues)
+                    {
+                        Func<InvariantDictionary<ConfigField>, bool> get = (config) => configField.Value.Contains(allowedValue);
+                        Action<InvariantDictionary<ConfigField>, bool> set = (config, val) =>
+                        {
+                            InvariantHashSet values = new InvariantHashSet(configField.Value);
+                            if (val && !values.Contains(allowedValue))
+                                values.Add(allowedValue);
+                            if (!val && values.Contains(allowedValue)) 
+                                values.Remove(allowedValue);
+
+                            if (values.Count == 0 && !configField.AllowBlank)
+                                values = configField.DefaultValues;
+
+                            configField.Value = values;
+                            valueProvider.SetValue(values);
+                        };
+                        configMenuIntegration.AddCheckbox($"{configName}.{allowedValue}", null, get, set);
+                    }
+                }
+                else
+                {
+                    // Whitelist + single value = drop down
+                    // Need an extra option when blank is allowed
+                    List<string> choices = new List<string>(configField.AllowValues);
+                    if (configField.AllowBlank)
+                        choices.Insert(0, "");
+
+                    Func<InvariantDictionary<ConfigField>, string> get = (config) => configField.Value.Count > 0 ? configField.Value.First() : "";
+                    Action<InvariantDictionary<ConfigField>, string> set = (config, val) =>
+                    {
+                        InvariantHashSet values = new InvariantHashSet(val);
+                        configField.Value = values;
+                        valueProvider.SetValue(values);
+                    };
+                    configMenuIntegration.AddDropdown(configName, null, choices.ToArray(), get, set);
+                }
+            }
+            else
+            {
+                // No whitelist = text field
+                Func<InvariantDictionary<ConfigField>, string> get = (config) => string.Join( ", ", configField.Value.ToArray());
+                Action<InvariantDictionary<ConfigField>, string> set = (config, val) =>
+                {
+                    InvariantHashSet values = configField.AllowMultiple ? this.ParseCommaDelimitedField(val) : new InvariantHashSet(val);
```

Comment:
Allowing commas in a value may cause odd effects, since Content Patcher often assumes comma = separate value. Is it possible to pass a filter to GMCM so prohibited characters can't be typed instead?

Reply:
Not at the moment. I could add a RegisterSimpleOptionFilter(IManifest manifest, string optionName, Func<string, string> filter). Would that work?

Reply:
I changed the logic to only save the first value instead if multiple values aren't allowed, which is good enough for now and should be reasonably intuitive.